### PR TITLE
Add support for Debian 9

### PR DESCRIPTION
**Pull request by ritwickdsouza**
References #6 . Creating new PR because made some mistake while sqaushing.

### CODE REVIEW

File: tasks/apt.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@
   notify: reload systemd
   with_items:
     - sudo
-    - python-software-properties
+    - software-properties-common
```
Comment (ritwickdsouza) [inline tasks/apt.yml]: @nemesisdesign `python-software-properties` is deprecated so replacing it with newer `software-properties-common`

File: tasks/apt.yml [hunk 2]
```diff
@@ -41,6 +41,12 @@
     - python3-dev
     - python-virtualenv
 
+- name: Install python3-pyparsing
+  when: ansible_distribution == "Debian" and ansible_distribution_version == "9"
+  apt: name={{ item }} state=latest
+  with_items:
+    - python3-pyparsing
```
Comment (ritwickdsouza) [inline tasks/apt.yml]: @nemesisdesign I think this should be sufficient. What do you think. ?

Comment (nemesisdesign) [inline tasks/apt.yml]: Are you sure `ansible_distribution_version == "9"` have you tested it?

File: tasks/apt.yml (hunk 1 above)
Comment (nemesisdesign) [inline tasks/apt.yml]: It could work. Have you tested it on a debian9 VM?

**Comment by nemesisdesign** [comment]
Installing on a clean debian9 (rc2) VM I get this issue with uwsgi, which seems the problem described by @zelda14 in #6:

```
[x86_64-linux-gnu-gcc -pthread] core/ssl.o
    core/ssl.c: In function ‘uwsgi_ssl_init’:
    core/ssl.c:17:9: warning: ‘OPENSSL_config’ is deprecated [-Wdeprecated-declarations]
             OPENSSL_config(NULL);
             ^~~~~~~~~~~~~~
    In file included from /usr/include/openssl/crypto.h:32:0,
                     from /usr/include/openssl/bio.h:20,
                     from /usr/include/openssl/conf.h:13,
                     from ./uwsgi.h:356,
                     from core/ssl.c:1:
    /usr/include/openssl/conf.h:92:1: note: declared here
     DEPRECATEDIN_1_1_0(void OPENSSL_config(const char *config_name))
     ^
    core/ssl.c: In function ‘uwsgi_ssl_info_cb’:
    core/ssl.c:26:24: error: dereferencing pointer to incomplete type ‘SSL {aka const struct ssl_st}’
                     if (ssl->s3) {
                            ^~
    core/ssl.c: In function ‘uwsgi_ssl_session_new_cb’:
    core/ssl.c:62:69: error: dereferencing pointer to incomplete type ‘SSL_SESSION {aka struct ssl_session_st}’
             if (uwsgi_cache_set2(uwsgi.ssl_sessions_cache, (char *) sess->session_id, sess->session_id_length, session_blob, len, uwsgi.ssl_sessions_timeout, 0)) {
                                                                         ^~
    core/ssl.c: In function ‘uwsgi_ssl_new_server_context’:
    core/ssl.c:408:46: warning: passing argument 2 of ‘SSL_CTX_sess_set_get_cb’ from incompatible pointer type [-Wincompatible-pointer-types]
                     SSL_CTX_sess_set_get_cb(ctx, uwsgi_ssl_session_get_cb);
                                                  ^~~~~~~~~~~~~~~~~~~~~~~~
    In file included from ./uwsgi.h:357:0,
                     from core/ssl.c:1:
    /usr/include/openssl/ssl.h:621:6: note: expected ‘SSL_SESSION * (*)(struct ssl_st *, const unsigned char *, int,  int *) {aka struct ssl_session_st * (*)(struct ssl_st *, const unsigned char *, int,  int *)}’ but argument is of type ‘SSL_SESSION * (*)(SSL *, unsigned char *, int,  int *) {aka struct ssl_session_st * (*)(struct ssl_st *, unsigned char *, int,  int *)}’
     void SSL_CTX_sess_set_get_cb(SSL_CTX *ctx,
          ^~~~~~~~~~~~~~~~~~~~~~~

    ----------------------------------------
Command "/usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-98ci_r46/uwsgi/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-2kra0hr4-record/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-build-98ci_r46/uwsgi/
```

**Comment by nemesisdesign** [comment]
I suggest you to proceed with the following steps:

- create a variable `openwisp2_uwsgi_package` in `defaults/main.yml` which defaults to `uwsgi` (don't forget to update the README by adding this new var in the relevant section with a short descriptive comment)
- change the line where the pip module is used to install uwsgi to use the new variable instead of the hardcoded `uwsgi` value
- before that task, use ansible `set_fact` module to modify the value of `openwisp2_uwsgi_package` to `http://http.debian.net/debian/pool/main/u/uwsgi/uwsgi_2.0.14+20170111.orig.tar.gz` only for debian9

**Comment by ritwickdsouza** [comment]
I am unable to generate the issue. Even on a clean install.

**Comment by nemesisdesign** [comment]
@ritwickdsouza, what image are you using? I used http://cdimage.debian.org/cdimage/stretch_di_rc2/amd64/iso-cd/debian-stretch-DI-rc2-amd64-netinst.iso

**Comment by ritwickdsouza** [comment]
I'm using https://atlas.hashicorp.com/remram/boxes/debian-9-i386 this box on Vagrant

**Comment by nemesisdesign** [comment]
@ritwickdsouza If I was you I would find a way to use the official debian iso when performing this kind of tests

**Comment by nemesisdesign** [comment]
@ritwickdsouza great. You are almost there. Keep insisting and you will have learned a lot.

**Comment by ritwickdsouza** [comment]
The uwsgi bug has been solved. https://github.com/unbit/uwsgi/issues/1395. 
Installed on fresh install of Debian Stretch RC2

**Comment by nemesisdesign** [comment]
@ritwickdsouza lucky ass! AHAH

File: tasks/apt.yml [hunk 3]
```diff
@@ -41,6 +41,12 @@
     - python3-dev
     - python-virtualenv
 
+- name: Install python3-pyparsing
+  when: ansible_distribution == "Debian"
```
Comment (nemesisdesign) [inline tasks/apt.yml]: isn't this missing a condition to specify that it should be executed only on debian 9?

Comment (ritwickdsouza) [inline tasks/apt.yml]: What is wrong with `ansible_distribution_version == '9'`

Comment (ritwickdsouza) [inline tasks/apt.yml]: or should i do `ansible_distribution_release == 'stretch'`

Comment (nemesisdesign) [inline tasks/apt.yml]: Not sure, could you find out? You could use the debug module of ansible to print the value of `ansible_distribution_release`.

**Comment by nemesisdesign** [comment]
@ritwickdsouza I noticed you push a change, what's your finding?

**Comment by ritwickdsouza** [comment]
@nemesisdesign It was supposed to be `9.0` instead of `9`

**Comment by nemesisdesign** [comment]
I am merging. , thanksI will test next week.

**Comment by nemesisdesign** [comment]
@ritwickdsouza if you can send also this news on the mailing list, something like: Experimental Debian 9 support in master branch

**Pull request merged**